src/gasPriceOracle/util.ts

```diff
@@ -1,6 +1,15 @@
-import { providers } from "ethers";
-import { BigNumber } from "../utils";
+import assert from "assert";
+import { type Chain, type PublicClient, createPublicClient, http, Transport } from "viem";
+import * as chains from "viem/chains";
```

Comment:
fwiw it seems like this would be a non-starter for the FE/API, so I'd probably import the specific chains we support as a follow-up change.

Reply:
I think this should be fine because the API will be using the `gasPriceOracle` and we don't care too much about the bundle size in that context